### PR TITLE
Fix #1576 - config file/module in GUNICORN_CMD_ARGS

### LINKED ISSUE #1576: --config doesn't work in GUNICORN_CMD_ARGS
Specifying `--config` in the `GUNICORN_CMD_ARGS` environment variable quietly fails as config file loading only happens when it is passed as an argument:
https://github.com/benoitc/gunicorn/blob/328e509260ae70de6c04c5ba885ee17960b3ced5/gunicorn/app/base.py#L137-L175

### PR DESCRIPTION
**Pull request by Code0x58**
I also sorted THANKS.md to follow CONTRIBUTING.md.

### CODE REVIEW

File: gunicorn/app/base.py [hunk 1]
```diff
@@ -147,26 +147,27 @@ def load_config(self):
             for k, v in cfg.items():
                 self.cfg.set(k.lower(), v)
 
-        if args.config:
-            self.load_config_from_file(args.config)
+        env_args = parser.parse_args(self.cfg.get_cmd_args_from_env())
+
+        if args.config or env_args.config:
```
Comment (tilgovi) [inline gunicorn/app/base.py]: Should one override the other rather than allowing both of these branches?

**Comment by Code0x58** [comment]
I can kind of understand why overriding would make sense, but inheriting feels more natural to me.

After looking at this more I've realised the hierarchy feels a bit weird (as it was before the patch) as options from the config file on the command line are overridden by arguments from the environment:
  app config ← ((env config file ← arg config file) | default config file) ← env options ← arg options

This also gives the issue that applications don't get augments from the environment, so for applications to get configured properly `Application.init` has to be after all of the config loading.

_Blerg._

**Comment by berkerpeksag** [comment]
Some comments:

* Please leave changes to THANKS file for another PR. You can add your name in this PR.
* There is no need to rename tests/config/test_cfg.py to tests/config/test_cfg1.py.

File: gunicorn/app/base.py (hunk 1 above)
Comment (berkerpeksag) [reply gunicorn/app/base.py]: I agree that `env_args.config` should override `args.config`. We should probably document this at http://docs.gunicorn.org/en/latest/settings.html#settings

**Comment by tilgovi** [comment]
I think this looks good to merge! Thank you so much, @Code0x58!

**Comment by tilgovi** [comment]
And thanks for your patience. I had my personal laptop melt down last weekend, so I've been a little bit offline more than usually.

**Comment by Code0x58** [comment]
Don't worry, I was delayed too as I was on a mobile for 5 days.

Is it worth me putting in a separate PR to either sort the THANKS, or remove the sorting part from CONTRIBUTING?

**Comment by tilgovi** [comment]
I'm fine to keep it sorted.

**Pull request merged**